### PR TITLE
Upgrade Helm to v3.6.3 and use apiextensions-application

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm#19819

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -5,33 +5,32 @@ go 1.16
...
+	github.com/giantswarm/app/v6 v6.1.0
 	github.com/giantswarm/appcatalog v0.6.0
-	github.com/giantswarm/apptest v0.12.0
+	github.com/giantswarm/apptest v1.0.0
 	github.com/giantswarm/backoff v0.2.0
 	github.com/giantswarm/errors v0.3.0
-	github.com/giantswarm/helmclient/v4 v4.8.0
-	github.com/giantswarm/k8sclient/v5 v5.12.0
+	github.com/giantswarm/helmclient/v4 v4.9.0
+	github.com/giantswarm/k8sclient/v6 v6.0.1-0.20211216105926-ea56d139d1fe
```
Comment (rossf7) [inline go.mod:15]: Needs https://github.com/giantswarm/k8sclient/pull/211

File: go.mod [hunk 2]
```diff
@@ -5,33 +5,32 @@ go 1.16
...
 	github.com/giantswarm/kubeconfig/v4 v4.1.0
-	github.com/giantswarm/microendpoint v0.2.0
-	github.com/giantswarm/microerror v0.3.0
+	github.com/giantswarm/microendpoint v0.3.0
+	github.com/giantswarm/microerror v0.4.0
 	github.com/giantswarm/microkit v0.2.2
-	github.com/giantswarm/micrologger v0.5.0
-	github.com/giantswarm/operatorkit/v5 v5.0.0
+	github.com/giantswarm/micrologger v0.6.0
+	github.com/giantswarm/operatorkit/v6 v6.0.1-0.20211216110045-41c3087cf2db
```
Comment (rossf7) [inline go.mod:22]: Needs https://github.com/giantswarm/operatorkit/pull/587

**Review by rossf7 — COMMENTED** [review]
Tested on gauss

**Comment by rossf7** [comment]
Tested on gauss

**Review by kubasobon — APPROVED** [review]
LGTMIIWFY

**Pull request merged**